src/main/java/practice/StreamPractice.java

```diff
@@ -1,20 +1,33 @@
 package practice;
 
-import java.util.Collections;
+import java.util.Arrays;
 import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 import model.Candidate;
+import model.Cat;
 import model.Person;
 
 public class StreamPractice {
+
+    private final Predicate<Integer> isOddNumber = n -> n % 2 != 0;
     /**
      * Given list of strings where each element contains 1+ numbers:
      * input = {"5,30,100", "0,22,7", ...}
      * return min integer value. One more thing - we're interested in even numbers.
      * If there is no needed data throw RuntimeException with message
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        int minEvenNumber = numbers.stream().map(s -> s.split(",")).flatMap(Arrays::stream)
+                .mapToInt(Integer::parseInt)
+                .filter(n -> !isOddNumber.test(n))
+                .min()
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list: "
+                        + numbers));
+        return minEvenNumber;
```

Comment:
```suggestion
        return numbers.stream().map(s -> s.split(",")).flatMap(Arrays::stream)
                .mapToInt(Integer::parseInt)
                .filter(n -> !isOddNumber.test(n))
                .min()
                .orElseThrow(() -> new RuntimeException("Can't get min value from list: "
                        + numbers));
```
the same for other methods